modules/pricer/src/main/java/com/opengamma/strata/pricer/calibration/MarketDataFxRateProvider.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.calibration;
+
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.currency.FxRate;
+import com.opengamma.strata.basics.currency.FxRateProvider;
+import com.opengamma.strata.basics.market.FxRateKey;
+import com.opengamma.strata.basics.market.MarketData;
+
+/**
+ * A provider of FX rates that takes its data from a {@link MarketData} instance.
+ */
+class MarketDataFxRateProvider implements FxRateProvider {
```

Comment:
`MarketData` could just implement `FxRateProvider`? While it makes it more specific, its not crazily more specific. It also allows other implementation possibilities, such as FX to be stored in an FxMatrix separate to the map of key to value.